File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/DTOS/CreateDTOS/CreatePromotionDTO.java
package app.aurafitbackend.DTOS.CreateDTOS;

import app.aurafitbackend.Beans.ProductVariant;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CreatePromotionDTO {
    private String name;
    private Integer discountPercent;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private ProductVariant productVariant;
}

File: aurafit-frontend/src/Models/DTOS/CreatePromotionDTO.ts
import { ProductVariant } from "../ProductVariant";

export class CreatePromotionDTO {

    name: string;
    discountPercent: number;
    startTime: Date;
    endTime: Date;
    productVariant: ProductVariant;

    constructor(name: string, discountPercent: number, startTime: Date, endTime: Date, productVariant: ProductVariant) {
        this.name = name;
        this.discountPercent = discountPercent;
        this.startTime = startTime;
        this.endTime = endTime;
        this.productVariant = productVariant;
    }

}